src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -3299,7 +3299,12 @@ void AuraEffect::HandleModMechanicImmunity(AuraApplication const* aurApp, uint8
     }
 
     if (apply && GetSpellInfo()->HasAttribute(SPELL_ATTR1_DISPEL_AURAS_ON_IMMUNITY))
-        target->RemoveAurasWithMechanic(mechanic, AURA_REMOVE_BY_DEFAULT, GetId());
```

Comment:
kay tested some more reports on this:

- hand of freedom does not remove frostnovaeffects
- Berserker Rage does not remove movemenimparing auras on activating

do you think we have to add a spellscript for these exceptions manually?